include/modem/location.h

```diff
@@ -0,0 +1,299 @@
+
+	/** Used positioning method. */
+	enum loc_method method;
+
+	union {
+		/** Current location, used with event LOC_EVT_LOCATION. */
+		struct loc_location location;
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+		/** A-GPS notification data frame used by GNSS to let the application know it
+		 *  needs new assistance data.
+		 */
+		struct nrf_modem_gnss_agps_data_frame request;
+#endif
+	};
+};
+
+/** GNSS configuration. */
+struct loc_gnss_config {
+	/**
+	 * @brief Timeout (in seconds), i.e. how long GNSS is allowed to run when trying to
+	 * acquire a fix.
+	 *
+	 * @details Note that this is not real time as experienced by the user.
+	 * Since GNSS cannot run while LTE is operating, the running time is
+	 * counted from the instant when GNSS is allowed to start. If LTE power saving mode (PSM)
+	 * is requested, GNSS waits until the modem enters PSM before starting up, thus maximizing
+	 * uninterrupted operating window and minimizing power consumption.
+	 */
+	uint16_t timeout;
+
+	/** Desired accuracy level. */
+	enum loc_accuracy accuracy;
+
+	/**
+	 * @brief If accuracy is set to LOC_ACCURACY_HIGH, instead of using the first fix, GNSS
+	 * is allowed to run for a longer time and produce the configured number fixes before the
+	 * library outputs the current location.
+	 *
+	 * @details This typically improves the location accuracy. If accuracy is set to
+	 * LOC_ACCURACY_NORMAL or LOC_ACCURACY_LOW this parameter has no effect.
+	 */
+	uint8_t num_consecutive_fixes;
+};
+
+/** LTE cellular positioning configuration. */
+struct loc_cellular_config {
+	/** Timeout (in seconds) on how long cellular positioning procedure can take. */
+	uint16_t timeout;
+
+	/** Used cellular positioning service. */
+	enum loc_service service;
+};
+
+/** WiFi positioning configuration. */
+struct loc_wifi_config {
+	/** Timeout (in seconds) on how long WiFi positioning procedure can take. */
+	uint16_t timeout;
+
+	/** Used WiFi positioning service. */
+	enum loc_service service;
+};
+
+/** Positioning method configuration. */
+struct loc_method_config {
+	/** Positioning method. */
+	enum loc_method method;
+	union {
+		/** Configuration for LOC_METHOD_CELLULAR. */
+		struct loc_cellular_config cellular;
+		/** Configuration for LOC_METHOD_GNSS. */
+		struct loc_gnss_config gnss;
+		/** Configuration for LOC_METHOD_WIFI. */
+		struct loc_wifi_config wifi;
+	};
+};
+
+/** Location request configuration. */
+struct loc_config {
+	/** Number of location methods in 'methods'. */
+	uint8_t methods_count;
+	/**
+	 * @brief Selected positioning methods and associated configurations in priority order.
+	 *
+	 * @details Index 0 has the highest priority. Number of methods is indicated in
+	 * 'methods_count'.
+	 */
+	struct loc_method_config *methods;
+	/**
+	 * @brief Position update interval in seconds.
+	 *
+	 * @details Set to 0 for a single position update. For periodic position updates
+	 * the valid range is 10...65535 seconds.
```

Comment:
True. I can change it to 32bit but I don't understand how to figure out if there is any limitation on the Zephyr timer side. And what do you mean with this: "It might not be possible to just use a single timer with a very large value." ?